### PR TITLE
Fix GenerateSequentialLayout implementation

### PR DESCRIPTION
**Pull request by josetr**

### CODE REVIEW

File: src/Generator/Generators/CSharp/CSharpSources.cs [hunk 1]
```diff
@@ -764,24 +767,63 @@ public override void GenerateClassSpecifier(Class @class)
...
-            Write($"internal {retType}");
-            if (field.Expression != null)
-            {
-                var fieldValuePrinted = field.Expression.CSharpValue(ExpressionPrinter);
-                Write($" = {fieldValuePrinted}");
+            if (fields.Count > 1)
+            { 
+                for (var i = 1; i < fields.Count; ++i)
+                {
+                    if (fields[i].Offset == fields[i - 1].Offset)
```
Comment (ddobrev) [inline src/Generator/Generators/CSharp/CSharpSources.cs:781]: Why is this necessary? Unless the type is a union which is already checked, fields at the same offset are impossible. If our AST reports this in any case, then it has a bug.

File: src/Generator/Generators/CSharp/CSharpSources.cs [hunk 2]
```diff
@@ -764,24 +767,63 @@ public override void GenerateClassSpecifier(Class @class)
...
+                if (sequentalLayout && fields[i].IsVTablePtr)
+                {
+                    var nativePointerSize = (Context.TargetInfo.PointerWidth / 8);
+                    var padding = 
+                        i + 1 < fields.Count
+                        ? fields[i + 1].Offset - field.Offset - nativePointerSize
+                        : (uint)@class.Layout.Size - nativePointerSize;
+
+                    if (padding > 0 && padding <= @class.Layout.Size)
+                        WriteLine($"public fixed byte {field.Name}Padding[{padding}];");
```
Comment (ddobrev) [inline src/Generator/Generators/CSharp/CSharpSources.cs:822]: I didn't get why this padding is necessary. It looks like a bug, a field missing from the AST.

File: src/Generator/Generators/CSharp/CSharpSources.cs (hunk 1 above)
Comment (josetr) [reply src/Generator/Generators/CSharp/CSharpSources.cs:781]: > fields at the same offset are impossible. 

```
class Time
{
public:
    // a & b will share the same offset
    union 
    {
        int a;
        double b;
    };
    int test;
};```

Comment (ddobrev) [reply src/Generator/Generators/CSharp/CSharpSources.cs:781]: Indeed they are – we just inline fields of empty types. Perhaps we need to reconsider this.

**Comment by ddobrev** [comment]
Merged at https://github.com/mono/CppSharp/commit/0d32121eee5b814493c86656f2f67e59ed21696a.
